### ISSUE TITLE
Variable selection

### ISSUE THREAD

**Issue by dongleihu** [privefl/bigstatsr#167]
Is there a way to extract the variables that LASSO selected?  When I run the command "summary(test, best.only = TRUE)" as described in the document of big_spLogReg (https://privefl.github.io/bigstatsr/reference/big_spLogReg.html), it shows that nb_var is 99.  Does that mean 99 variables were selected?  How would I extract those 99 variables?

Thanks,
Donglei

**Comment by privefl** [comment]
You have the `$beta` (for both `X` and `covar`) so that you can check the number and positions of non-zero values in there.

**Comment by dongleihu** [comment]
When K=3 as in the example, there are 3 sets of beta.  Which one should I use?  Is it the average of all 3?  Thanks!

**Comment by privefl** [comment]
There are only two models in the summary, corresponding to the two values of alpha in the example.
You can choose the one with the smallest validation loss.

**Comment by dongleihu** [comment]
privefl, thank you for your response! Yes, there are 2 results corresponding to 2 alpha values: test[[1]] and test[[2]]. "test[[2]]" is for alpha=1 which produced the smallest validation loss. "test[[2]]" itself has 3 sets of beta: test[[2]][[1]]$beta, test[[2]][[2]]$beta, and test[[2]][[3]]$beta. Each set of beta has a different number of non-zero values. I wonder how I can combine these 3 sets of beta. Thank you!

**Comment by privefl** [comment]
Yes, you can take the average of those. This is what you actually get in the summary().

**Comment by dongleihu** [comment]
@privefl Thank you!

**Comment by privefl** [comment]
The last 3 are for covar.

**Comment by dongleihu** [comment]
> The last 3 are for covar.

Thank you!
